pkg/proxy/proxy.go

```diff
@@ -241,15 +244,14 @@ func uploadToBlobStorage(ctx context.Context, cacheID int) (*DockerGHAUploadCach
 		finalBuffer.Write(cacheEntry.Chunks[offset].Content)
 	}
 
-	if cacheEntry.BackendReserveResponse.Provider == ProviderS3 {
+	switch cacheEntry.BackendReserveResponse.Provider {
```

Comment:
Do you see any value in shifting to a provider specific layers like we do in `backend-core` instead of a switch case?

Reply:
We will just add Azure after this. I don't think the amount of provider code is significant enough to refactor this. Switch should work for a long time.